### ISSUE TITLE
How to use this program?

### ISSUE THREAD

**Issue by magnascale** [ohwgiles/NetworkManager-f5vpn#1]
Hi.
I installed `networkmanager-f5vpn` from AUR on Arch Linux.
I'm expecting to see in my network manager ability to add F5 connection.
But I don't have this.
![image](https://user-images.githubusercontent.com/61244077/96578351-bc3b3000-12dd-11eb-9922-30cfb73fb337.png)

**How to use this program?**
Thank you.

**Comment by ohwgiles** [comment]
You might need to restart NetworkManager after installing the plugin: `sudo systemctl restart NetworkManager`.

Is that KDE? I have only tested this under xfce with `nm-connection-editor`, perhaps something extra is required to make it appear in the KDE menu.

**Comment by magnascale** [comment]
> You might need to restart NetworkManager after installing the plugin: `sudo systemctl restart NetworkManager`.
> 
> Is that KDE? I have only tested this under xfce with `nm-connection-editor`, perhaps something extra is required to make it appear in the KDE menu.

I did restart of whole system, it didn't appear.
Yes, KDE 5.20.
I was able to add F5VPN using `nm-connection-editor`.
To connect to F5VPN I have host, username and password.
In this config window I can't find where I can specify username and password.
Can you, please, show screenshot of your configuration, how it looks.
Thank you.

**Comment by ohwgiles** [comment]
> Yes, KDE 5.20.
> I was able to add F5VPN using nm-connection-editor.

OK so it looks like there is something missing for KDE

> To connect to F5VPN I have host, username and password.
> In this config window I can't find where I can specify username and password.

Different servers have different demands for credentials. For instance, my usual one has username, password and OTP code. Just use the hostname in the configuration dialog, it will prompt you for the missing credentials when it tries to connect.

In the case where an OTP (or similar) is requested, it needs to be re-entered whenever there is no valid session key. For just a username/password this is tedious - they could be stored. The plugin doesn't do that yet, but it could be improved in this regard.

**Comment by magnascale** [comment]
What I did:
1. `nm-connection-editor`, than selected F5VPN, entered name and host
![image](https://user-images.githubusercontent.com/61244077/96968645-f0029980-1519-11eb-83ac-518eaa9fc356.png)
2. Using CLI of Network Manager I connect to VPN 
```
nmcli con up V5VPN 
Ошибка: сбой активации соединения: No valid secrets
Hint: use 'journalctl -xe NM_CONNECTION=058ecc76-079c-4b7b-8d71-be68cd0dcf36 + NM_DEVICE=wlp5s0' to get more details.
```

It says: `Error: connection activation failed: No valid secrets `

3. This is output from `journalctl -xe NM_CONNECTION=058ecc76-079c-4b7b-8d71-be68cd0dcf36 + NM_DEVICE=wlp5s0`
```
окт 23 10:27:13 arch NetworkManager[474]: <info>  [1603438033.6033] dhcp4 (wlp5s0): option requested_subnet_mask => '1'
окт 23 10:27:13 arch NetworkManager[474]: <info>  [1603438033.6033] dhcp4 (wlp5s0): option requested_time_offset => '1'
окт 23 10:27:13 arch NetworkManager[474]: <info>  [1603438033.6034] dhcp4 (wlp5s0): option requested_wpad       => '1'
окт 23 10:27:13 arch NetworkManager[474]: <info>  [1603438033.6034] dhcp4 (wlp5s0): option routers              => '192.168.88.1'
окт 23 10:27:13 arch NetworkManager[474]: <info>  [1603438033.6034] dhcp4 (wlp5s0): option subnet_mask          => '255.255.255.0'
окт 23 10:27:13 arch NetworkManager[474]: <info>  [1603438033.6035] dhcp4 (wlp5s0): state changed extended -> extended
окт 23 10:28:57 arch NetworkManager[474]: <info>  [1603438137.9170] vpn-connection[0x55b80df80760,058ecc76-079c-4b7b-8d71-be68cd0dcf36,"F5VPN",0]: Started the VPN service, PID 371134
окт 23 10:28:57 arch NetworkManager[474]: <info>  [1603438137.9344] vpn-connection[0x55b80df80760,058ecc76-079c-4b7b-8d71-be68cd0dcf36,"F5VPN",0]: Saw the service appear; activating connection
окт 23 10:28:58 arch NetworkManager[474]: <error> [1603438138.0034] vpn-connection[0x55b80df80760,058ecc76-079c-4b7b-8d71-be68cd0dcf36,"F5VPN",0]: Failed to request VPN secrets #3: No agents were available for this request.
окт 23 10:28:58 arch NetworkManager[474]: <info>  [1603438138.0304] vpn-connection[0x55b80df80760,058ecc76-079c-4b7b-8d71-be68cd0dcf36,"F5VPN",0]: VPN plugin: state changed: stopped (6)
```

**Any idea?**
Thanks for the help.

**Comment by ohwgiles** [comment]
I think this is again a KDE integration issue.

This plugin contains a gtk "authentication dialog" which should be shown at this point, but it seems it is not being launched in your environment.

Are you running `nm-applet`? If not, can you see whether running that during the connect fixes the issue?

You might also try `nmcli --ask c u <connection name>`

**Comment by magnascale** [comment]
```
nm-applet
zsh: command not found: nm-applet
```
I don't have `nm-applet` program. And I don't see it in repository.

```
nmcli --ask c u F5VPN
Ошибка: сбой активации соединения: No valid secrets
Hint: use 'journalctl -xe NM_CONNECTION=058ecc76-079c-4b7b-8d71-be68cd0dcf36 + NM_DEVICE=wlp5s0' to get more details.
```
Same issue as before.
But I received notification
![image](https://user-images.githubusercontent.com/61244077/97156203-0cebd680-177f-11eb-8ca9-36f10362883e.png)
which says something like `Passwords needed for connection were not provided`
Don't know how to fix it.

**Comment by ohwgiles** [comment]
> I don't have nm-applet program. And I don't see it in repository.

It is `network-manager-applet` on Arch. Please try installing and running it.

I don't have time at the moment to implement KDE/plasma integration. This workaround should hopefully bring in the minimal GTK dependencies you need to make the GTK integration work.

**Comment by magnascale** [comment]
@ohwgiles 
I have this window, and it doesn't have any fields to input data.
![image](https://user-images.githubusercontent.com/61244077/97281069-bbf2e580-1845-11eb-8d86-65db93a3f638.png)

I understand, I will try some other solution.
Thanks for help!

**Comment by ohwgiles** [comment]
OK now it looks like you are at the same place as #2

**Comment by ohwgiles** [comment]
Considering this closed in da79b3f
